linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/LinkerdHeaders.scala

```diff
@@ -449,7 +449,14 @@ object Headers {
           req.headerMap -= k
         }
       }
-      service(req)
+      service(req).map { rsp =>
+        rsp.headerMap.get(Err.Key) match {
+          case Some(_) =>
+            rsp.clearContent()
```

Comment:
We could also potentially strip the `l5d-err`, `l5d-success-class`, and `l5d-retryable` headers off the response here.  What do you think?